### PR TITLE
Set up Redux for Assessment state

### PR DESCRIPTION
**Pull request by MikeD-ZOE**
# Description

[Builds on PR #364 / move-assessment-dtos]

Getting Redux up and running in the app to manage Assessment state. Mainly figuring out how to structure code in a modular way that reflects the Service approach.

Just one place where the clean mapping is icky, creating the rootReducer needs to reach into assessment for the reducer.

* I've used an enum type to encapsulate the Action Types, and enforced types on method parameters and return values.

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [X] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist

- [ ] I have updated mockServer
- [X] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up

_Is there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/core/assessment/state/reducers.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+const initialState: Partial<AssessmentInfosRequest> = {};
+
+const reducer = (state = initialState, action: AssessmentAction): Partial<AssessmentInfosRequest> => {
+  switch (action.type) {
+    case ActionTypes.UPDATE_ASSESSMENT:
+      return {
+        ...state,
+        ...action.payload,
+      };
```
Comment (julienlavigne) [inline src/core/assessment/state/reducers.ts:12]: Purely FYI, for a later PR. See Redux guidelines of using a library like immer.js to manage immutable state

File: src/core/assessment/state/types.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { AssessmentInfosRequest } from '../dto/AssessmentInfosRequest';
+
+export enum ActionTypes {
+  UPDATE_ASSESSMENT = 'UPDATE_ASSESSMENT',
+}
```
Comment (julienlavigne) [inline src/core/assessment/state/types.ts:5]: Redux guideline is to name action past tense as events.

also FYI typesafe actions will avoid some boilerplate there

File: src/core/state/root.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { combineReducers } from 'redux';
+
+import assessment from '../assessment/state/reducers';
+
+export default combineReducers({ assessment });
```
Comment (julienlavigne) [inline src/core/state/root.ts:5]: I know you were worried about the root reducer reaching out to the assessment feature but I think that's ok.

File: src/core/assessment/state/reducers.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { AssessmentInfosRequest } from '../dto/AssessmentInfosRequest';
+import { ActionTypes, AssessmentAction } from './types';
+
+const initialState: Partial<AssessmentInfosRequest> = {};
+
+const reducer = (state = initialState, action: AssessmentAction): Partial<AssessmentInfosRequest> => {
+  switch (action.type) {
+    case ActionTypes.UPDATE_ASSESSMENT:
```
Comment (julienlavigne) [inline src/core/assessment/state/reducers.ts:8]: FYI typesafe actions will help there. The current version of your code doesn't have strong typing on the action variable, which means you will need to cast things. At that point, typescript won't be able to help (you will literally tell it this is a X, trust me)

File: src/core/assessment/state/types.ts (hunk 2 above)
Comment (MikeD-ZOE) [reply src/core/assessment/state/types.ts]: Switched action to `ASSESSMENT_UPDATED`

**Comment by MikeD-ZOE** [comment]
Changing from draft to review. I'm happy it's not breaking the current app.

**Pull request merged**